### ISSUE TITLE
when querying by id, gen yields the same result over and over

### ISSUE THREAD

**Issue by rhaksw** [dmarx/psaw#46]
```
from psaw import PushshiftAPI

api = PushshiftAPI()

gen = api.search_submissions(ids=['byefwi'])
for x in gen:
  print(x.id)
```

When I run this on psaw 0.0.7, I get,

```
byefwi
byefwi
byefwi
...
```

I have not looked into the cause. I was able to workaround this for my purposes by querying reddit instead of Pushshift.

**Comment by phiob** [comment]
I ran also into this problem. I could solve it by providing a limit which reflects the count of ids I'm asking the API for.

**Comment by ghost** [comment]
Just to confirm, this is definitely still a thing but the workaround of setting limit equal to the len of the list of ids seems to work.

**Comment by dmarx** [comment]
Fixed in commit 467ee54. Will be included in v0.0.8 to be pushed to PyPI shortly.
